pkg/apis/release/v1alpha1/release_types.go

```diff
@@ -74,11 +74,21 @@ type Release struct {
 type ReleaseSpec struct {
 	// Apps describes apps used in this release.
 	Apps []ReleaseSpecApp `json:"apps"`
+
 	// +kubebuilder:validation:MinItems=1
 	// Components describes components used in this release.
 	Components []ReleaseSpecComponent `json:"components"`
+
 	// Date that the release became active.
 	Date *metav1.Time `json:"date"`
+
+	// +kubebuilder:validation:Optional
+	// +nullable
+	// EndOfLifeDate is the date and time when support for a tenant cluster using
+	// this release ends. This may not be set at the time of release creation
+	// and can be specififed later.
+	EndOfLifeDate *metav1.Time `json:"endOfLifeDate"`
```

Comment:
```suggestion
	EndOfLifeDate *metav1.Time `json:"endOfLifeDate,omitempty"`
```
This hasn't really been documented yet, but I think it's better to use `omitempty` whenever we make a field optional.

Reply:
Which only means that Go will not write this attribute when creating a JSON representation, in the case the value is `nil`. Do we consider this cleaner?

Reply:
It's cleaner, but there are more good reasons here https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
I think this is relevant to the percent spot instances field you were working on recently too.

Reply:
There we removed `omitempty` again because it caused us trouble. But that was an integer field and it was not declared as a pointer. So `omitempty` resulted in the attribute being omitted whenever the value was zero. Here the case is different.

Reply:
Added `omitempty` here